docs/source/dependencies.md

```diff
@@ -0,0 +1,92 @@
+
+The capability (in case of the leapp tool, leapp framework, and snactor) is
+called **leapp-framework-dependencies**. Anytime the *outer* dependencies are
+changed, the value of capability has to be incremented by one - so the
+information about the change is provided to other leapp related packages.
+
+This metapackage can be simply replaced by different one, that will handle
+outer dependencies of the target system. That means, that the metapackage
+with dependencies for a target system is not part of this git repository
+and it is completely on a leapp repository using the leapp tool (or framework)
+to provide such package in case it is needed.
+
+## Why do I need to have a special way to define a dependency
+
+One possible use of the Leapp framework is to do the in-place upgrade (IPU).
+In that case, it is expected Leapp needs to be executed in two different
+versions of the system - it starts with the source system and then continues
+with the target (upgraded) system. It is quite common that some packages are
+renamed between those two systems (or in case of leapp repository, you would
+like to use technology on the new system that is not provided on the original
+one).
+In such cases, it is hard to satisfy dependencies on both systems - when you
+want to proceed with the upgrade process with the same Leapp packages.
+
+As the solution we are using the metapackage that contains those dependencies.
+
+## What to do when dependencies needs to be changed
+
+It's easy to change *outer dependencies* for Leapp. Open the
+`packaging/leapp.spec` file and change dependencies as it is needed for
+`%package deps`. The right place is pretty highlighted (you cannot miss it):
+
+```spec
+##################################################
+# Real requirements for the leapp HERE
+##################################################
+...
+Requires: findutils
+Requires: your-dependency
+```
+
+And do not forget to increment value of `leapp-framework-dependencies`:
+
+```spec
+# IMPORTANT: everytime the requirements are changed, increment number by one
+# - same for Provides in deps subpackage
+Requires: leapp-framework-dependencies = 1
+```
+
+**IMPORTANT** Do the same thing for all `Requires` and `Provides` of the
+`leapp-framework-dependencies` capability in the SPEC file. Ensure that value
+is consistent across the SPEC file.
+
+### Some more unusual steps
+
+If you add *outer dependencies* in Leapp framework you must add this in
+[Leapp repository](https://github.com/oamg/leapp-repository) metapackage too,
+this is necessary to making sure Leapp will run on the target system in the
+in-place upgrade.
+
+For now, we have only RHEL 8 as target system, so you need to edit
+[leapp-repository/packaging/leapp-el7toel8-deps.spec](https://github.com/oamg/leapp-repository/blob/master/packaging/leapp-el7toel8-deps.spec)
+ and add your dependency, remember to change the value of
+`leapp-framework-dependencies` to match with your framework value.
+
```

Comment:
I think that this information should be part of just leapp-repository repo. It is not neccessary here. We should mentioned that there it is one official leapp repository, which needs to reflect the changes (paragraph above) but that's all. Keep in mind that this framework can be used by anyone for whatever purposes. Another thing is, that we do not want to modify documentation everytime we will add something new into leapp repository. Keep it more "abstract". In case that people will look at the leapp-repository in case of change in leapp, it's pretty ok. Maybe link to the leapp repository documentation for handling of deps would be good idea as well.

Reply:
Removed